### PR TITLE
Use runtime gauge condition in GH

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Switches GH to using a runtime gauge condition specified in the input file.

There'll be a few more PRs that reduce the amount of redundant work done in the DH gauge, but this is the final piece for runtime gauge conditions and getting rid of the rollon gauge.

Only last two commits are new.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveGhBinaryBlackHole.hpp [hunk 1]
```diff
@@ -244,15 +248,23 @@ struct EvolutionMetavars {
...
       tmpl::list<
-          gr::Tags::Lapse<DataVector>,
-          gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
-          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
           gr::Tags::SpacetimeMetric<volume_dim, Frame::Inertial, DataVector>,
           GeneralizedHarmonic::Tags::Pi<volume_dim, Frame::Inertial>,
           GeneralizedHarmonic::Tags::Phi<volume_dim, Frame::Inertial>,
-          GeneralizedHarmonic::Tags::GaugeH<volume_dim, Frame::Inertial>,
-          GeneralizedHarmonic::Tags::SpacetimeDerivGaugeH<volume_dim,
-                                                          Frame::Inertial>,
```
Comment (knelli2) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveGhBinaryBlackHole.hpp]: Will I still be able to observe the gauge and it's derivative? I see the compute tag is added to `non_tensor_compute_tags` below, but the gauge and deriv tags are removed from the `observe_fields` list.

File: src/Evolution/Executables/GeneralizedHarmonic/GeneralizedHarmonicBase.hpp [hunk 2]
```diff
@@ -209,7 +210,19 @@ struct GeneralizedHarmonicTemplateBase<
 
   using observe_fields = tmpl::append<
       tmpl::push_back<
-          analytic_solution_fields, gr::Tags::Lapse<DataVector>,
+          analytic_solution_fields,
+          gr::Tags::SpatialMetric<volume_dim, frame, DataVector>,
```
Comment (knelli2) [inline src/Evolution/Executables/GeneralizedHarmonic/GeneralizedHarmonicBase.hpp]: If you add the gauge and deriv back to the BBH executable, can you also add them here?

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/TimeDerivativeTerms.hpp [hunk 3]
```diff
@@ -62,10 +62,25 @@ struct TimeDerivativeTermsImpl<
           comoving_magnetic_field_one_form,
       const typename ::Tags::deriv<GhGradientTags, tmpl::size_t<3_st>,
                                    Frame::Inertial>::type&... gh_gradients,
-      const typename GhArgTags::type&... gh_args,
-      const typename ValenciaArgTags::type&... valencia_args) {
+      // db::const_item_type<GhArgTags, tmpl::list<>>... gh_args,
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/GhValenciaDivClean/TimeDerivativeTerms.hpp]: remove?

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
Rebased and squashed. Thanks for the review!

**Review by knelli2 — COMMENTED** [review]
Couple comments. If you decide to make the changes you can squash directly.

**Comment by nilsdeppe** [comment]
Done and added `GaugeH` to the Kerr input file so as to keep checking for it in the future

**Review by knelli2 — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**